modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -0,0 +1,645 @@
+#include <sstream>
+
+#include "darknet_io.hpp"
+
+namespace darknet {
+
+    template<typename T>
+    std::string toString(const T &v)
+    {
+        std::ostringstream ss;
+        ss << v;
+        return ss.str();
+    }
+
+    template<typename T>
+    T getParam(const std::map<std::string, std::string> &params, const std::string param_name, T init_val)
+    {
+        std::map<std::string, std::string>::const_iterator it = params.find(param_name);
+        if (it != params.end()) {
+            std::stringstream ss(it->second);
+            ss >> init_val;
+        }
+        return init_val;
+    }
+
+    class setLayersParams {
+
+        NetParameter *net;
+        int layer_id;
+        std::string last_layer;
+
+    public:
+        setLayersParams(NetParameter *_net, std::string _first_layer = "data") :
+            net(_net), layer_id(0), last_layer(_first_layer)
+        {}
+
+        void setLayerBlobs(int i, std::vector<cv::Mat> blobs)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams &params = net->layers[i].layerParams;
+            params.blobs = blobs;
+        }
+
+        void setLastLayerName(std::string layer_name)
+        {
+            if(net->layers.size() > 0)
+                net->layers.rbegin()->layer_name = layer_name;
+        }
+
+        cv::dnn::experimental_dnn_v1::LayerParams getParamConvolution(int kernel_h, int kernel_w, int pad_h, int pad_w,
+            int stride_h, int stride_w, int filters_num, int channels_num)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams params;
+
+            params.name = "Convolution-name";
+            params.type = "Convolution";
+
+            params.set<int>("kernel_h", kernel_h);
+            params.set<int>("kernel_w", kernel_w);
+            params.set<int>("pad_h", pad_h);
+            params.set<int>("pad_w", pad_w);
+            params.set<int>("stride_h", stride_h);
+            params.set<int>("stride_w", stride_w);
+            params.set<int>("dilation_h", 1);
+            params.set<int>("dilation_w", 1);
+
+            params.set<bool>("bias_term", false);	// true only if(BatchNorm == false)
+            params.set<int>("num_output", filters_num);
+            params.set<int>("group", 1);	// ngroups
+
+            params.set<int>("adj_h", 0);
+            params.set<int>("adj_w", 0);
+
+            return params;
+        }
+
+
+        void setConvolution(int kernel_h, int kernel_w, int pad_h, int pad_w, int stride_h, int stride_w,
+            int filters_num, int channels_num, int use_batch_normalize, int use_relu)
+        {
+            cv::dnn::experimental_dnn_v1::LayerParams conv_param =
+                getParamConvolution(kernel_h, kernel_w, pad_h, pad_w, stride_h, stride_w,
+                    filters_num, channels_num);
+
+            darknet::LayerParameter lp;
+            std::string layer_name = toString(layer_id);
+            if (use_batch_normalize || use_relu) layer_name = "conv_" + layer_name;
```

Comment:
It's better to name layers using type prefix every time. Moreover some of layers just named with numbers and it's hard to debug them.